task/common/identifier.go

```diff
@@ -12,38 +12,46 @@ import (
 	"github.com/aohorodnyk/uid"
 )
 
-type Identifier string
+type Identifier struct {
+	name string
+	salt string
+}
 
 var ErrWrongIdentifier = errors.New("wrong identifier")
 
 const (
 	maximumLongLength = 50
 	shortLength       = 16
+	nameLength        = maximumLongLength-shortLength-uint32(len("tpi---"))
```

Comment:
why `tpi---` ?

Reply:
`len("tpi---")` is the length of the `tpi` prefix plus the three hyphens separating fields `{1}`, `{2}` and `{3}`. I thought it would be less confusing than a hardcoded `6` but probably it isn't the case. 🤷🏼‍♂️

Reply:
This should be in a comment above `nameLength` - the perfect example of explaining _why_ :)

I'd probably just define `maximumLength` as a combination of `nameLength` and `shortLength`.

Reply:
> _This should be in a comment above nameLength - the perfect example of explaining why_

Comments, comments, ... 😈 

> _I'd probably just define maximumLength as a combination of nameLength and shortLength._

Sounds good, although I'm a bit reluctant to do that. I wanted the total length limits to be really obvious, as they're the important numbers. E.g. resource X names on cloud Y can have a maximum length of Z characters, and our names are below that number.

Reply:
I see where you're coming from, but still - `maximumLength` is a function of `(shortLength, nameLength)`, not the other way around. And if we want to restrict `maximumLength` from exceeding a specific value, go's tests are a great way to do that.

Reply:
> _`maximumLength` is a function of `(shortLength, nameLength)`, not the other way around_

Why can't it be the other way around? 🤔 If you want to avoid operations other than addition, you're right, but... would it be any easier to reason about in that way?





> _And if we want to restrict `maximumLength` from exceeding a specific value, go's tests are a great way to do that._

Isn't that a bit more convoluted than the current approach? 🤔